sources/packages/api/src/route-controllers/student/student.students.controller.ts

```diff
@@ -199,4 +200,15 @@ export class StudentStudentsController extends BaseController {
       fileMetadata,
     );
   }
+
+  /**
+   * Get the student information that represents the profile.
+   * @returns student profile information.
+   */
+  @Get()
+  async getStudentProfile(
+    @UserToken() userToken: StudentUserToken,
+  ): Promise<StudentProfileAPIOutDTO> {
+    return this.studentControllerService.getStudentProfile(userToken.studentId);
+  }
```

Comment:
I think you need to throw a "not found" exception when a student's profile is not found.

Reply:
For the student, if this code is reached the student was already validated by the guards and validators and we are sure that the student account is present.

Reply:
Yeah, makes sense.